src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/EditProceedingsAndCostsSectionController.java

```diff
@@ -755,4 +822,379 @@ public String proceedingsActionConfirmPost(
+   * @param scopeLimitationFlow the scope limitation flow data, retrieved from
+   *                            the session.
+   * @param scopeLimitationDetails the submitted details of the scope limitation.
+   * @param model the {@link Model} object for passing attributes to the view.
+   * @param bindingResult the result of the validation process.
+   * @return the name of the view to render or a redirect path.
+   */
+  @PostMapping("/application/proceedings/scope-limitations/{action}/details")
+  public String scopeLimitationDetailsPost(
+      @SessionAttribute(APPLICATION) final ApplicationDetail application,
+      @SessionAttribute(PROCEEDING_FLOW_FORM_DATA)
+      final ProceedingFlowFormData proceedingFlow,
+      @SessionAttribute(SCOPE_LIMITATION_FLOW_FORM_DATA)
+      final ScopeLimitationFlowFormData scopeLimitationFlow,
+      @ModelAttribute("scopeLimitationDetails")
+      final ScopeLimitationFormDataDetails scopeLimitationDetails,
+      final Model model,
+      final BindingResult bindingResult) {
+
+    scopeLimitationDetailsValidator.validate(scopeLimitationDetails, bindingResult);
+
+    if (bindingResult.hasErrors()) {
+      populateScopeLimitationDropdown(model, application, proceedingFlow);
+      model.addAttribute(PROCEEDING_FLOW_FORM_DATA, proceedingFlow);
+      model.addAttribute(SCOPE_LIMITATION_FLOW_FORM_DATA, scopeLimitationFlow);
+      return "application/proceedings-scope-limitations-details";
+    }
+
+    scopeLimitationFlow.setScopeLimitationDetails(scopeLimitationDetails);
+    populateScopeLimitationDetails(model, application, proceedingFlow, scopeLimitationFlow);
+
+    return "redirect:/application/proceedings/scope-limitations/confirm";
+  }
+
+  /**
+   * Populates the dropdown for scope limitations in the model.
+   *
+   * <p>Utilizes application and proceeding flow data to determine the criteria
+   * for filtering scope limitations. Adjusts criteria for emergency applications
+   * and retrieves a sorted list of scope limitation details for dropdown
+   * population.</p>
+   *
+   * @param model the {@link Model} object for passing attributes to the view.
+   * @param application the application details, used to set criteria.
+   * @param proceedingFlow the proceeding flow data, used to set criteria.
+   */
+  private void populateScopeLimitationDropdown(
+      final Model model,
+      final ApplicationDetail application,
+      final ProceedingFlowFormData proceedingFlow) {
+
+    final ScopeLimitationDetail criteria = new ScopeLimitationDetail()
+        .categoryOfLaw(application.getCategoryOfLaw().getId())
+        .matterType(proceedingFlow.getMatterTypeDetails().getMatterType())
+        .proceedingCode(proceedingFlow.getProceedingDetails().getProceedingType())
+        .levelOfService(proceedingFlow.getFurtherDetails().getLevelOfService());
+
+    if (EMERGENCY_APPLICATION_TYPE_CODES.contains(application.getApplicationType().getId())) {
```

Comment:
This appears twice, so maybe refactor into a method?

Reply:
refactored